applications/tari_validator_node/src/p2p/services/committee_state_sync.rs

```diff
@@ -158,16 +180,63 @@ impl CommitteeStateSync {
                 // inventory.push(sync_vn_shard.into());
                 substate_count += 1;
             }
+            let mut new_missing_blocks = Vec::new();
+            let mut block_count = 0;
+            // We always want to sync from our tip to anything new.
+            missing_blocks.push((*tip.id(), None));
+            for (from, to) in missing_blocks {
+                let mut vn_blocks_stream = sync_vn_rpc_client
+                    .sync_blocks(tari_validator_node_rpc::proto::rpc::BlockSyncRequest {
+                        start_block_id: from.as_bytes().to_vec(),
+                        end_block_id: match to {
+                            Some(to) => to.as_bytes().to_vec(),
+                            None => vec![],
+                        },
+                        epoch: epoch.as_u64(),
+                    })
+                    .await?;
+                info!(target: LOG_TARGET, "🌍 Syncing blocks...");
+                let mut last_synced_block = to;
+                while let Some(resp) = vn_blocks_stream.next().await {
+                    let msg = resp?;
+                    let block: Block<CommsPublicKey> = msg
+                        .block
+                        .ok_or(CommitteeStateSyncError::InvalidStateSyncData(anyhow::anyhow!(
+                            "No block"
+                        )))?
+                        .try_into()
+                        .map_err(CommitteeStateSyncError::InvalidStateSyncData)?;
+                    self.shard_store.with_write_tx(|tx| block.justify().save(tx))?;
+                    self.shard_store.with_write_tx(|tx| block.insert(tx))?;
```

Comment:
These should run in a single transaction
```suggestion
                    self.shard_store.with_write_tx(|tx| {
                       block.justify().save(tx)?;
                       block.insert(tx)
                    })?;
```